src/wp-includes/block-template-utils.php

```diff
@@ -761,38 +819,24 @@ function _build_block_template_result_from_post( $post ) {
 	$origin           = get_post_meta( $parent_post->ID, 'origin', true );
 	$is_wp_suggestion = get_post_meta( $parent_post->ID, 'is_wp_suggestion', true );
 
-	$template                 = new WP_Block_Template();
-	$template->wp_id          = $post->ID;
-	$template->id             = $theme . '//' . $parent_post->post_name;
-	$template->theme          = $theme;
-	$template->content        = $post->post_content;
-	$template->slug           = $post->post_name;
-	$template->source         = 'custom';
-	$template->origin         = ! empty( $origin ) ? $origin : null;
-	$template->type           = $post->post_type;
-	$template->description    = $post->post_excerpt;
-	$template->title          = $post->post_title;
-	$template->status         = $post->post_status;
-	$template->has_theme_file = $has_theme_file;
-	$template->is_custom      = empty( $is_wp_suggestion );
-	$template->author         = $post->post_author;
-	$template->modified       = $post->post_modified;
-
-	if ( 'wp_template' === $parent_post->post_type && $has_theme_file && isset( $template_file['postTypes'] ) ) {
-		$template->post_types = $template_file['postTypes'];
-	}
-
-	if ( 'wp_template' === $parent_post->post_type && isset( $default_template_types[ $template->slug ] ) ) {
-		$template->is_custom = false;
-	}
+	$additional_fields = array(
+		'id'             => $theme . '//' . $parent_post->post_name,
+		'theme'          => $theme,
```

Comment:
On the note of `_build_block_template_object_from_wp_post_object` being more opinionated: I'd move any logic that computes one based on others into that helper. This minimizes the risk of "clashing" arguments (e.g. `id` => `theme//template`, `theme` => `other-theme//template` ???), removes the burden of computing those from the caller, and reduces the number of arguments that need to be passed to the helper.

```suggestion
		'theme'          => $theme,
```

Reply:
Yeah, that makes sense. I am assuming you mean the `id` field. We wouldn't be able to move the `$theme` field easily into the helper since its value is derived from different sources depending on where its called.

Reply:
Yeah, the `$id`. The `$theme` will need to continue to be an argument.

I'm currently looking into https://github.com/WordPress/wordpress-develop/pull/3533 to better understand the `$parent_post` logic.

Reply:
https://github.com/WordPress/wordpress-develop/pull/6278/commits/f85f3ce56aa641efc490fd52773b753ec59f5622